faster_whisper/transcribe.py

```diff
@@ -377,143 +378,9 @@ def load_vad_model(self, vad_onset=0.500, vad_offset=0.363):
-        return word_timings
-
-    def combine_words(self, metadata, response):
-        combined_segments = []
-
-        for meta, res in zip(metadata, response):
-            word_timestamps = res["word_timestamps"]
-            segment_texts = []
-            segment_index = 0
-            current_segment = meta["segments"][segment_index]
-            current_text = []
-            current_word_timestamps = []
-            current_start = current_segment[0]
-
-            for idx, word_info in enumerate(word_timestamps):
-                word_start, word_end, word_text = (
-                    word_info["start"],
-                    word_info["end"],
-                    word_info["word"],
-                )
-
-                # Move to the next segment if the word is outside the current segment
-                while (
-                    word_start >= current_segment[1]
-                    and segment_index < len(meta["segments"]) - 1
-                ):
-                    # Save the completed segment
-                    if current_text:
-                        segment_texts.append(
-                            {
-                                "start": current_start,
-                                "end": current_segment[1],
-                                "text": "".join(current_text),
-                                "word_timestamps": current_word_timestamps,
-                                "avg_logprob": res["avg_logprob"],
-                                "no_speech_prob": res["no_speech_prob"],
-                            }
-                        )
-                    segment_index += 1
-                    current_segment = meta["segments"][segment_index]
-                    current_start = current_segment[0]
-                    current_text = []
-                    current_word_timestamps = []
-
-                # Add word to the current segment text
-                if word_start >= current_segment[0] and word_end <= current_segment[1]:
-                    current_text.append(word_text)
-                    current_word_timestamps.append(word_info)
-
-            # Save the final segment
-            if current_text:
-                segment_texts.append(
-                    {
-                        "start": current_start,
-                        "end": current_segment[1],
-                        "text": "".join(current_text),
-                        "word_timestamps": current_word_timestamps,
-                        "avg_logprob": res["avg_logprob"],
-                        "no_speech_prob": res["no_speech_prob"],
-                    }
-                )
-
-            combined_segments.extend(segment_texts)
-        return combined_segments
-
     def transcribe(
         self,
-        audio: Union[str, np.ndarray],
+        audio: Union[str, torch.Tensor, np.ndarray],
         vad_segments: Optional[List[dict]] = None,
         batch_size: int = 16,
         num_workers: int = 0,
```

Comment:
Remove `num_workers` from arguments and use the one from original `WhisperModel` object. It can be set when the model is loaded with `WhisperModel`. Currently, it does not work if you provide `num_workers` to 1 or greater.

Reply:
when setting whisper model's `num_workers` to 0 it's extremely slow, so I suggest hardcoding it to 0 regardless of other parameters

Reply:
There are still redundant variable checks that can be avoided.  Set `self._num_workers=0` in __init__ and just call:
```
dataloader = torch.utils.data.DataLoader(
            dataset, num_workers=self._num_workers, batch_size=batch_size, collate_fn=stack
        )
```
inside the iterator.